### PR TITLE
Typescript magic

### PR DESCRIPTION
**Pull request by topce**
update libs 
small improvements in typescript typing

`export type Key = 'a0' | `${File}${Rank}`;`

`export function callUserFunction<T  extends (...args: any) => void>(
  f: T | undefined,
  ...args: Parameters<T>
): void {
  if (f) setTimeout(() => f(args), 1);
}`

fix compile time errors 
add 
 premove: false,
not sure if should be true ?

commit dist files

### CODE REVIEW

File: src/board.ts [hunk 1]
```diff
@@ -3,10 +3,11 @@ import { pos2key, key2pos, opposite, distanceSq, allPos, computeSquareCenter } f
...
 
-export type Callback = (...args: any[]) => void;
-
-export function callUserFunction(f: Callback | undefined, ...args: any[]): void {
-  if (f) setTimeout(() => f(...args), 1);
+export function callUserFunction<T  extends (...args: any) => void>(
+  f: T | undefined,
+  ...args: Parameters<T>
+): void {
+  if (f) setTimeout(() => f(args), 1);
```
Comment (niklasf) [inline src/board.ts:10]: Wow, really cool features. Is `...args` -> `args` intended here?

Comment (topce) [reply src/board.ts:10]: Thank you
Yes it is intended
https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype

Comment (niklasf) [reply src/board.ts:10]: Ok :+1:

First time I see type annotations changing semantics.

Comment (niklasf) [reply src/board.ts:10]: Actually ... looking at the generated code, it does not seem right: https://github.com/ornicar/chessground/pull/157/files#diff-930f9993ca95523e8be4573892f7254508010d0e452834b60403d3e6e3b96761R185-R188

Comment (topce) [reply src/board.ts:10]: You are right , I will doubecheck

**Comment by topce** [comment]
@niklasf indeed it was error 
I commit fix to call apply 
`if (f) setTimeout(() => f.apply(null,args), 1); `
if try to call it like before 
`if (f) setTimeout(() => f(...args), 1); `
there is TS compile error 
(parameter) args: Parameters<T>
Type 'Parameters<T>' must have a '[Symbol.iterator]()' method that returns an iterator.ts(2488)

still if you want to get rid of error and do not call apply
it could be ignored 
`// @ts-ignore
   if (f) setTimeout(() => f(...args), 1); `

**Comment by topce** [comment]
create bug [41728](https://github.com/microsoft/TypeScript/issues/41728) in TypeScript repo

**Comment by topce** [comment]
@niklasf  with last commit it should work same way like before 
with f(...args) it look like when you call spread operator in function in typescript is not same as if 
you call apply in javascript it is same but not in TS
trick was to change any -> any[]
ugly but it works better than other solutions ;-)

**Comment by niklasf** [comment]
Looks good, thanks!

**Pull request merged**